File: mvvm-java/domain/src/test/java/com/wasisto/githubuserfinder/domain/NoOpLogger.java
package com.wasisto.githubuserfinder.domain;

import com.wasisto.githubuserfinder.common.Logger;

public class NoOpLogger implements Logger {

    @Override
    public void trace(String message) {}

    @Override
    public void trace(String message, Throwable throwable) {}

    @Override
    public void debug(String message) {}

    @Override
    public void debug(String message, Throwable throwable) {}

    @Override
    public void info(String message) {}

    @Override
    public void info(String message, Throwable throwable) {}

    @Override
    public void warn(String message) {}

    @Override
    public void warn(String message, Throwable throwable) {}

    @Override
    public void error(String message) {}

    @Override
    public void error(String message, Throwable throwable) {}

    @Override
    public void fatal(String message) {}

    @Override
    public void fatal(String message, Throwable throwable) {}
}

File: mvvm-kotlin/ui/src/test/java/com/wasisto/githubuserfinder/android/NoOpLogger.kt
package com.wasisto.githubuserfinder.android

import com.wasisto.githubuserfinder.common.Logger

object NoOpLogger : Logger {

    override fun trace(message: String) {}

    override fun trace(message: String, throwable: Throwable) {}

    override fun debug(message: String) {}

    override fun debug(message: String, throwable: Throwable) {}

    override fun info(message: String) {}

    override fun info(message: String, throwable: Throwable) {}

    override fun warn(message: String) {}

    override fun warn(message: String, throwable: Throwable) {}

    override fun error(message: String) {}

    override fun error(message: String, throwable: Throwable) {}

    override fun fatal(message: String) {}

    override fun fatal(message: String, throwable: Throwable) {}
}